File: packages/dartvel_cli/bin/dartvel.dart
#!/usr/bin/env dart

import 'dart:io';
import 'package:dartvel_cli/dartvel_impl.dart' as impl;

Future<void> main(List<String> cliArgs) async {
  try {
    await impl.main(cliArgs);
  } catch (error, stacktrace) {
    stderr.writeln(error);
    stderr.writeln(stacktrace);

    exit(1);
  }
}

File: npm/dartvel_cli/bin/dartvel.js
#!/usr/bin/env node

'use strict';

// dartvel_cli is an alias for dartvel_dev.
//
// The CLI is published on pub.dev as dartvel_cli and on npm as dartvel_dev,
// because `dartvel` was already taken on pub.dev by an unrelated package. That
// left `npm i -g dartvel_cli` installing nothing, which is the name people
// reach for first. This package exists so both work and resolve to one
// implementation -- a second copy of the launcher would be a second thing to
// keep in step.

const { run } = require('dartvel_dev');

run(process.argv.slice(2))
  .then((code) => process.exit(code))
  .catch((error) => {
    console.error('Error running dartvel:', error.message);
    process.exit(1);
  });
